app/helpers/session.rb

```diff
@@ -0,0 +1,11 @@
+def current_user
+  @current_user ||= User.find_by(id: session[:user_id])
+end
+
+def logged_in?
+  !!user_loggedin
```

Comment:
Heads up: there is no `user_loggedin` method defined anywhere in your code base. Thus this line will always raise an error. Here you will want to coerce `current_user` into its boolean equivalent:
```ruby
!!current_user
```